src/test/consensus/Validations_test.cpp

```diff
@@ -0,0 +1,611 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+#include <BeastConfig.h>
+#include <ripple/beast/clock/manual_clock.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/consensus/Validations.h>
+
+#include <type_traits>
+#include <vector>
+
+
+namespace ripple {
+namespace test {
+
+class Validations_test : public beast::unit_test::suite
+{
+    using clock_type = beast::abstract_clock<std::chrono::steady_clock> const;
+    //--------------------------------------------------------------------------
+    // Basic type wrappers for validation types
+
+    // Represents a ledger sequence number
+    struct Seq
+    {
+        explicit Seq(std::size_t sIn) : s{sIn}
+        {
+        }
+
+        Seq() : s{0}
+        {
+        }
+
+        operator std::size_t() const
+        {
+            return s;
+        }
+
+        std::size_t s;
+    };
+
+    // Represents a unique ledger identifier
+    struct ID
+    {
+        explicit ID(std::size_t idIn) : id{idIn}
+        {
+        }
+
+        ID() : id{0}
+        {
+        }
+
+        int
+        signum() const
+        {
+            return id == 0 ? 0 : 1;
+        }
+
+        operator std::size_t() const
+        {
+            return id;
+        }
+
+        template <class Hasher>
+        friend void
+        hash_append(Hasher& h, ID const& id)
+        {
+            using beast::hash_append;
+            hash_append(h, id.id);
+        }
+
+        std::size_t id;
```

Comment:
I'm happy to change, but in principle this test should be independent of the bitwidth of `std::size_t`.  If a platform has a different width, its doing an extra test for us! ;)